chains/orchestrator-relays/runtime/dancelight/src/bridge_to_ethereum_config.rs

```
@@ -345,6 +333,239 @@ where
+        InboundQueuePalletInstance,
+    )>,
+);
+
+impl<T, EthereumLocation, EthereumNetwork, InboundQueuePalletInstance> MessageProcessor
+    for NativeContainerTokensProcessor<
+        T,
+        EthereumLocation,
+        EthereumNetwork,
+        InboundQueuePalletInstance,
+    >
+where
+    T: snowbridge_pallet_inbound_queue::Config
+        + pallet_ethereum_token_transfers::Config
+        + snowbridge_pallet_system::Config
+        + pallet_xcm::Config,
+    <T as frame_system::Config>::RuntimeEvent: From<RuntimeEvent>,
+    EthereumLocation: Get<Location>,
+    EthereumNetwork: Get<NetworkId>,
+    InboundQueuePalletInstance: Get<u8>,
+{
+    fn can_process_message(channel: &Channel, envelope: &Envelope) -> bool {
+        // Validate channel and gateway
+        if !GatewayAndChannelValidator::<T>::validate_gateway_and_channel(channel, envelope) {
+            log::warn!(
+                "NativeContainerTokensProcessor::can_process_message: invalid gateway or channel"
+            );
+            return false;
+        }
+
+        // Try decoding the message and check if the destination owns the token being transferred
+        if let Some(token_data) =
+            NativeTokenTransferData::decode_native_token_message(&envelope.payload)
+        {
+            // Check if the asset is registered on EthereumSystem
+            if let Some(token_location) = Self::get_token_location(&token_data.token_id) {
+                Self::validate_destination_owns_token(&token_location, &token_data.destination)
+            } else {
+                log::error!(
+                    "NativeContainerTokensProcessor::can_process_message: token location not found for token_id: {:?}",
+                    token_data.token_id
+                );
+                false
+            }
+        } else {
+            false
+        }
+    }
+
+    fn process_message(_channel: Channel, envelope: Envelope) -> DispatchResult {
+        if let Some(token_data) =
```

We might want to throw an explicit error here in case the payload has not been decoded properly; right now we consider it as a success. Or If we want to avoid  stalled pipe, maybe add a similar error log at least before `Ok(())`.

---

If the payload is not decoded properly, `process_message` will be unreachable, as the payload it's also checked inside `can_process_message`, but I can add an explicit error here as well just in case.

---

Also done in 97f01bb6bb4e1cd3fc59d70dd5f5170f3c25d0d7.